src/Middleware/SocialAuthMiddleware.php

```diff
@@ -228,16 +225,16 @@ protected function _handleCallbackAction(ServerRequest $request): Response
 
         $user->unset($config['fields']['password']);
 
-        if (!$config['userEntity']) {
-            $user = $user->toArray();
-        }
-
-        $event = $this->dispatchEvent(self::EVENT_AFTER_IDENTIFY, ['user' => $user]);
+        $event = $this->dispatchEvent(self::EVENT_AFTER_IDENTIFY, ['user' => $user, 'profile' => $profile]);
```

Comment:
The associated profile entity would be already set to `$user->social_profile`, so there's no need to pass it separately again.